### ISSUE TITLE
Hamburger menu doesn’t work for keyboard and high-contrast mode users

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#7327]
The hamburger markup is not a link nor button, which breaks keyboard support, and styling for high-contrast mode users.

See for example with high-contrast mode – the button should be styled with an "active" color:

![Current](https://user-images.githubusercontent.com/877585/125082245-e5dff580-e0be-11eb-8568-a9d22de320bd.png)

---

The solution for this is to change the markup so this toggle is a `<button>` element.
